oiosaml/src/test/java/dk/gov/oio/saml/session/SessionDestroyListenerTest.java

```diff
@@ -0,0 +1,93 @@
+                .setIdpEntityID(TestConstants.IDP_ENTITY_ID)
+                .setIdpMetadataUrl(TestConstants.IDP_METADATA_URL)
+                .setSessionHandlerFactoryClassName(TestSessionHandlerFactory.class.getName())
+                .setKeystoreLocation("sp.pfx")
+                .setKeystorePassword("Test1234")
+                .setKeyAlias("1")
+                .build();
+
+        OIOSAML3Service.init(configuration);
+
+    }
+
+    @DisplayName("Test that current session is logged out after execution")
+    @Test
+    public void logoutCurrentSession() throws Exception {
+        SessionDestroyListener sessionDestroyListener = new SessionDestroyListener();
+
+        // mock session with state: not logged in at any NSIS level
+        HttpSession session = Mockito.mock(HttpSession.class);
+
+        HttpSessionEvent httpSessionEvent = Mockito.mock(HttpSessionEvent.class);
+        Mockito.when(httpSessionEvent.getSession()).thenReturn(session);
+
+        AssertionWrapper assertionWrapper = Mockito.mock(AssertionWrapper.class);
+        Mockito.when(assertionWrapper.getNsisLevel()).thenReturn(null);
+
+        SessionHandler sessionHandler = OIOSAML3Service.getSessionHandlerFactory().getHandler();
+        Mockito.when(sessionHandler.getAssertion(session)).thenReturn(assertionWrapper);
+        Mockito.when(sessionHandler.isAuthenticated(session)).thenReturn(true);
+
+        // mock request
+        HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+        Mockito.when(request.getSession()).thenReturn(session);
+
+        sessionDestroyListener.sessionCreated(httpSessionEvent);
+
+        sessionDestroyListener.sessionDestroyed(httpSessionEvent);
+
+        Mockito.verify(sessionHandler,Mockito.times(1)).logout(session,assertionWrapper);
+    }
+
+    @DisplayName("Test that session is ignored if not logged in")
+    @Test
+    public void ignoreMissingSession() throws Exception {
+        SessionDestroyListener sessionDestroyListener = new SessionDestroyListener();
+
+        // mock session with state: not logged in at any NSIS level
+        HttpSession session = Mockito.mock(HttpSession.class);
+
+        HttpSessionEvent httpSessionEvent = Mockito.mock(HttpSessionEvent.class);
+        Mockito.when(httpSessionEvent.getSession()).thenReturn(session);
+
+        SessionHandler sessionHandler = OIOSAML3Service.getSessionHandlerFactory().getHandler();
+        Mockito.when(sessionHandler.isAuthenticated(session)).thenReturn(false);
+
+        // mock request
+        HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+        Mockito.when(request.getSession()).thenReturn(session);
+
+        sessionDestroyListener.sessionCreated(httpSessionEvent);
+
+        sessionDestroyListener.sessionDestroyed(httpSessionEvent);
+
+        Mockito.verify(sessionHandler,Mockito.never()).logout(Mockito.eq(session),Mockito.any(AssertionWrapper.class));
```

Comment:
Testen bliver endnu stærkere hvis du asserter på any session i stedet for eq(session)